docs/content/integrations/dbt/using-dbt-with-dagster/downstream-assets.mdx

```diff
@@ -13,119 +13,112 @@ description: Dagster can orchestrate dbt alongside other technologies.
+```shell
+pip install plotly
+```
 
-Here's a quick refresher of what our graph of assets currently looks like:
+## Step 2: Create the order_count_chart asset
 
-<Image
-alt="Asset graph in the Dagster UI, showing materialized assets"
-src="/images/integrations/dbt/using-dbt-with-dagster/upstream-assets-asset-graph-materialized.png"
-width={1091}
-height={896}
-/>
+You've added upstream assets to your data pipeline, but nothing downstream - until now. In this step, you'll create a Dagster asset called `order_count_chart` that uses the data in the `customers` dbt model to computes a plotly chart of the number of orders per customer.
+
+Like the `raw_customers` asset that we added in the [previous section](/integrations/dbt/using-dbt-with-dagster/upstream-assets#create-upstream-dagster-assets), we'll put this asset in our `definitions.py` file, inside the `jaffle_dagster` directory.
 
 To add the `order_count_chart` asset:
 
-1. In `/tutorial_template/tutorial_dbt_dagster/assets/__init__.py`, replace the imports section with the following:
+1. Replace the imports section with the following:
 
-   ```python
+   ```python file=integrations/dbt/tutorial/downstream_assets/definitions.py startafter=start_imports endbefore=end_imports
+   import os
+   from pathlib import Path
+
+   import duckdb
    import pandas as pd
    import plotly.express as px
-   from dagster_dbt import load_assets_from_dbt_project
-
-   from dagster import AssetIn, MetadataValue, asset, file_relative_path
+   from dagster_dbt import (
+       DbtCliResource,
+       build_schedule_from_dbt_selection,
+       dbt_assets,
+       get_asset_key_for_model,
+   )
+
+   from dagster import Definitions, MetadataValue, OpExecutionContext, asset
    ```
 
-2. At the bottom of the file, add the following:
+   This adds an import for plotly, as well as <PyObject module="dagster_dbt" object="get_asset_key_for_model" /> and <PyObject module="dagster" object="MetadataValue" />, which we'll use in our asset.
+
+2. Before `defs = Definitions(`, add the definition for the `order_count_chart` asset:
 
-   ```python
-   @asset(ins={"customers": AssetIn(key_prefix=["jaffle_shop"])})
-   def order_count_chart(context, customers: pd.DataFrame) -> None:
+   ```python file=integrations/dbt/tutorial/downstream_assets/definitions.py startafter=start_downstream_asset endbefore=end_downstream_asset
+   @asset(deps=get_asset_key_for_model([jaffle_shop_dbt_assets], "customers"))
+   def order_count_chart(context):
+       # read the contents of the customers table into a Pandas DataFrame
+       connection = duckdb.connect(os.fspath(duckdb_database_path))
+       customers = connection.sql("select * from customers").df()
+
+       # create a plot of number of orders by customer and write it out to an HTML file
        fig = px.histogram(customers, x="number_of_orders")
        fig.update_layout(bargap=0.2)
```

Comment:
Is `update_layout` really necessary?

Reply:
very possibly not, but this was the code that was here before, so my bias is to leave it for now to minimize the number of moving pieces